src/test/java/com/google/step/servlets/PromoCodeServletTest.java

```diff
@@ -1,12 +1,128 @@
+import static org.mockito.Mockito.anyString;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.Optional;
+
+import com.google.step.youtube.YoutubeInfoScraper;
+import com.google.step.youtube.PromoCode;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.youtube.model.Channel;
+import com.google.api.services.youtube.model.ChannelListResponse;
+import com.google.api.services.youtube.model.ChannelContentDetails;
+import com.google.api.services.youtube.model.ChannelContentDetails.RelatedPlaylists;
+import com.google.api.services.youtube.model.PlaylistItem;
+import com.google.api.services.youtube.model.PlaylistItemSnippet;
+import com.google.api.services.youtube.model.PlaylistItemListResponse;
+import com.google.api.services.youtube.model.ResourceId;
+import com.google.api.services.youtube.YouTube.Channels;
+import com.google.api.services.youtube.YouTube.PlaylistItems;
+import com.google.api.services.youtube.YouTube;
+
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.runners.MockitoJUnitRunner;
 
-// @RunWith(JUnit4.class)
+@RunWith(MockitoJUnitRunner.class)
 public final class PromoCodeServletTest {
-    // TODO(margaret): Implementation coming
-}
+        
+    private PromoCodeServlet servlet;
+
+    private static final String CHANNEL_ID_NONEXISTENT = "CHANNEL_ID_NONEXISTENT";
+    private static final String UPLOAD_ID_NONEXISTENT = "UPLOAD_ID_NONEXISTENT";
+    private static final String CHANNEL_ID = "CHANNEL_ID";
+    private static final String UPLOAD_ID = "UPLOAD_ID";
+    private static final String VIDEO_ID = "VIDEO_ID";
+    private static final String IOEXCEPTION = "IOEXCEPTION";
+    private static final long MOCK_DATE = 0L;
+
+    @Mock
+    HttpServletRequest request;
+ 
+    @Mock
+    HttpServletResponse response;
+
+    @Mock
+    YoutubeInfoScraper infoScraper;
+
+    @Before
+    public void setup() throws IOException {
+        //MockitoAnnotations.initMocks(this);
+        
+        infoScraper = mock(YoutubeInfoScraper.class);
+        servlet = new PromoCodeServlet(infoScraper);
+    }
+
+    @Test
+    public void incorrectChannelIdRequest() throws ServletException, IOException {
+        when(request.getParameter("formInput")).thenReturn(CHANNEL_ID_NONEXISTENT);
+        
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
 
+        when(response.getWriter()).thenReturn(pw);
+        when(infoScraper.scrapeChannelUploadPlaylist(anyString())).thenReturn(Optional.empty());
```

Comment:
anyString() -> CHANNEL_ID_NONEXISTENT. Since this test is for the incorrectChannelIdRequest